app/src/lib/stores/app-store.ts

```diff
@@ -5058,30 +5077,52 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
-  /**
-   *  Note: **drops the existing desktop stash** on that branch (if there is one)
-   */
-  private async createStash(repository: Repository, branchName: string) {
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _createStashAndDropPreviousEntry(
+    repository: Repository,
+    branchName: string
+  ) {
     if (!enableStashing()) {
       return
     }
 
-    // get the previous stash before we create a new one
-    const previousStash = await getLastDesktopStashEntryForBranch(
+    const previousStashEntry = await getLastDesktopStashEntryForBranch(
       repository,
       branchName
     )
 
-    await createDesktopStashEntry(repository, branchName)
-
-    if (previousStash !== null) {
-      await dropDesktopStashEntry(repository, previousStash.stashSha)
+    if (previousStashEntry !== null) {
+      await dropDesktopStashEntry(repository, previousStashEntry.stashSha)
       log.info(
-        `Dropped stash '${previousStash.stashSha}' associated with ${
-          previousStash.branchName
+        `Dropped stash '${previousStashEntry.stashSha}' associated with ${
+          previousStashEntry.branchName
         }`
       )
     }
+
+    await createDesktopStashEntry(repository, branchName)
+  }
+
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _moveChangesToBranchAndCheckout(
+    repository: Repository,
+    branchToCheckout: string
+  ) {
+    if (!enableStashing()) {
+      return
+    }
+
+    await createDesktopStashEntry(repository, branchToCheckout)
```

Comment:
Could we wrap this in a `performFailableOperation` and only continue with the checkout if the operation succeeds? If we don't I'm assuming the app will crash rather than show an error message if creating the stash fails.